### ISSUE TITLE
A potential bug that self.model.training could be True in self.predict()

### ISSUE THREAD

**Issue by WenjieDu** [WenjieDu/PyPOTS#641]
### 1. System Info

pypots v0.12

### 2. Information

- [x] The official example scripts
- [ ] My own created scripts

### 3. Reproduction

During model training, if users interrupt the training procedure, like pressing ctrl+c, (this is very likely for many reasons, e.g., setting too large epochs), then run model predict()

### 4. Expected behavior

PyPOTS applied decorator @torch.no_grad() to predict() in PR #612 and removed `self.model.eval()` in predict(). 
If the training procedure gets interrupted, the model module may still be in the training state and attribute `training` is True, and this could result in unexpected operations in models core module, e.g. SAITS below

https://github.com/WenjieDu/PyPOTS/blob/47080186912db2e58dee4663cb58ecd97144c90a/pypots/imputation/saits/core.py#L94-L114

For loss calculation, it may be fine because we have stopped gradient calc with @torch.no_grad(). But it will result in potential bugs if some additional steps are put here in future model implementations.
